File: LeetCode_ProblemSet/Go/Problem238.go
package main 

func productExceptSelf(nums []int) []int {
	prefixProduct := make([]int, len(nums))
	suffixProduct := make([]int, len(nums))
	answer := make([]int, len(nums))
	prefixProduct[0] = 1
	suffixProduct[len(nums) - 1] = 1

	for i := 1; i < len(nums); i++ {
		prefixProduct[i] = prefixProduct[i - 1] * nums[i - 1]
	}

	for i := len(nums) - 2; i > -1; i-- {
		suffixProduct[i] = suffixProduct[i + 1] * nums[i + 1]
	}

	for i := 0; i < len(nums); i++ {
		answer[i] = prefixProduct[i] * suffixProduct[i]
	}
 
	return answer
}

File: LeetCode_ProblemSet/Problem238.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        prefixProduct = [1 for _ in range(len(nums))]
        suffixProduct = [1 for _ in range(len(nums))]
        
        for i in range(1, len(nums)):
            prefixProduct[i] = prefixProduct[i - 1] * nums[i - 1]
            
        for i in range(len(nums) - 2, -1, -1):
            suffixProduct[i] = suffixProduct[i + 1] * nums[i + 1]
            
        return [suffixProduct[i] * prefixProduct[i] for i in range(len(nums))]